jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -40,22 +44,42 @@
    * @throws IOException if extraction fails
    */
   public static void extract(Path source, Path destination) throws IOException {
-    String canonicalDestination = destination.toFile().getCanonicalPath();
+    extract(source, destination, false);
+  }
 
+  /**
+   * Extracts a tarball to the specified destination.
+   *
+   * @param source the tarball to extract
+   * @param destination the output directory
+   * @param enableReproducibleTimestamps whether or not reproducible timestamps should be used
+   * @throws IOException if extraction fails
+   * @throws IllegalStateException when reproducible timestamps are enabled but the target root used
+   *     for extracting the tar contents is not empty
+   */
+  public static void extract(Path source, Path destination, boolean enableReproducibleTimestamps)
+      throws IOException {
+    boolean isEmptyDestination =
+        Files.isDirectory(destination) && destination.toFile().list().length == 0;
```

Comment:
Let's do this computation only when necessary. That is, doing this inside the `if` statement. Perhaps
```java
if (enableReproducibleTimestamps && Files.isDirectory(destination) && destination.toFile().list.length() != 0) {
  throw new ...
}
```
(`isDirectory()` implies `exists()`.)

Reply:
Ah this is much better, thanks